test/built-ins/Array/prototype/concat/concat-array-likes-1.js

```diff
@@ -0,0 +1,677 @@
+// Copyright 2014 the V8 project authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flags: --harmony-arrays --harmony-classes
+
+/*---
+es6id: 22.1.3.1_3
+description: >
+    Array.prototype.concat: Symbol.isConcatSpreadable & array-likes
+---*/
+
+(function testArrayConcatArity() {
+  "use strict";
+  assert.deepEquals(1, Array.prototype.concat.length);
+})();
+
+
+(function testArrayConcatNoPrototype() {
+  "use strict";
+  assert.deepEquals(void 0, Array.prototype.concat.prototype);
+})();
+
+
+(function testArrayConcatDescriptor() {
+  "use strict";
```

Comment:
We were using explicit strict-mode in order to be able to test concat-spreadable Arguments objects, and in V8 there are three flavours of Arguments objects.